test_conformance/half/Test_vStoreHalf.cpp

```diff
@@ -208,211 +210,27 @@ CheckD(cl_uint jid, cl_uint tid, void *userInfo)
-    {
-        x *= MAKE_HEX_FLOAT(0x1.0p24f, 0x1L, 24);
-        return (cl_ushort)((int) x | sign);
-    }
-
-    u.u &= 0xFFFFE000U;
-    u.u -= 0x38000000U;
-
-    return (u.u >> (24-11)) | sign;
+static cl_ushort float2half_rtz( float f ) {
+    return cl_half_from_float(f, CL_HALF_RTZ);
 }
 
-static cl_ushort
-float2half_rtp( float f )
-{
-    union{ float f; cl_uint u; } u = {f};
-    cl_uint sign = (u.u >> 16) & 0x8000;
-    float x = fabsf(f);
-
-    //Nan
-    if( x != x )
-    {
-        u.u >>= (24-11);
-        u.u &= 0x7fff;
-        u.u |= 0x0200;      //silence the NaN
-        return u.u | sign;
-    }
-
-    // overflow
-    if( f > MAKE_HEX_FLOAT(0x1.ffcp15f, 0x1ffcL, 3) )
-        return 0x7c00;
-
-    if( f <= MAKE_HEX_FLOAT(-0x1.0p16f, -0x1L, 16) )
-    {
-        if( f == -INFINITY )
-            return 0xfc00;
-
-        return 0xfbff;
-    }
-
-    // underflow
-    if( x < MAKE_HEX_FLOAT(0x1.0p-24f, 0x1L, -24) )
-    {
-        if( f > 0 )
-            return 1;
-        return sign;
-    }
-
-    // half denormal
-    if( x < MAKE_HEX_FLOAT(0x1.0p-14f, 0x1L, -14) )
-    {
-        x *= MAKE_HEX_FLOAT(0x1.0p24f, 0x1L, 24);
-        int r = (int) x;
-        r += (float) r != x && f > 0.0f;
-
-        return (cl_ushort)( r | sign);
-    }
-
-    float g = u.f;
-    u.u &= 0xFFFFE000U;
-    if( g > u.f )
-        u.u += 0x00002000U;
-    u.u -= 0x38000000U;
-
-    return (u.u >> (24-11)) | sign;
+static cl_ushort float2half_rtp( float f ) {
+    return cl_half_from_float(f, CL_HALF_RTP);
 }
 
-
-static cl_ushort
-float2half_rtn( float f )
-{
-    union{ float f; cl_uint u; } u = {f};
-    cl_uint sign = (u.u >> 16) & 0x8000;
-    float x = fabsf(f);
-
-    //Nan
-    if( x != x )
-    {
-        u.u >>= (24-11);
-        u.u &= 0x7fff;
-        u.u |= 0x0200;      //silence the NaN
-        return u.u | sign;
-    }
-
-    // overflow
-    if( f >= MAKE_HEX_FLOAT(0x1.0p16f, 0x1L, 16) )
-    {
-        if( f == INFINITY )
-            return 0x7c00;
-
-        return 0x7bff;
-    }
-
-    if( f < MAKE_HEX_FLOAT(-0x1.ffcp15f, -0x1ffcL, 3) )
-        return 0xfc00;
-
-    // underflow
-    if( x < MAKE_HEX_FLOAT(0x1.0p-24f, 0x1L, -24) )
-    {
-        if( f < 0 )
-            return 0x8001;
-        return sign;
-    }
-
-    // half denormal
-    if( x < MAKE_HEX_FLOAT(0x1.0p-14f, 0x1L, -14) )
-    {
-        x *= MAKE_HEX_FLOAT(0x1.0p24f, 0x1L, 24);
-        int r = (int) x;
-        r += (float) r != x && f < 0.0f;
-
-        return (cl_ushort)( r | sign);
-    }
-
-    u.u &= 0xFFFFE000U;
-    if( u.f > f )
-        u.u += 0x00002000U;
-    u.u -= 0x38000000U;
-
-    return (u.u >> (24-11)) | sign;
+static cl_ushort float2half_rtn( float f ) {
+    return cl_half_from_float(f, CL_HALF_RTN);
 }
 
+static cl_ushort double2half_rte( double f );
+static cl_ushort double2half_rtz( double f );
+static cl_ushort double2half_rtp( double f );
+static cl_ushort double2half_rtn( double f );
+
```

Comment:
What's the reason for not using `cl_half_from_double` for these ones?

Reply:
I'd forgotten that you'd added them as well :). Will do.